numpy/tests/test_reloading.py

```diff
@@ -36,3 +36,62 @@ def test_novalue():
         assert_equal(repr(np._NoValue), '<no value>')
         assert_(pickle.loads(pickle.dumps(np._NoValue,
                                           protocol=proto)) is np._NoValue)
+
+
+def test_py2_can_load_py3_pickle_with_dtype_field_names():
```

Comment:
This isn't the right file for these tests - `test_reloading` is for testing `importlib.reload` specifically. 

I think this belongs in either `test_multiarray.TestPickling` or `test_dtype.TestPickling`.

If you put it in the latter, then I'd pickle just `expected_dtype`, and not use `expected_data` at all.